PJ05/src/MessageSystem.java

```diff
@@ -30,6 +30,16 @@ public User getUser(String name) throws UserNotFoundException {
         throw new UserNotFoundException();
     }
 
+    public User getUser(UUID find_uuid) throws UserNotFoundException {
+        for (UUID uuid : userDatabase.uuids()) {
+            User user = userDatabase.get(uuid);
+            if (user.uuid.equals(find_uuid)) {
+                return user;
+            }
+        }
+        throw new UserNotFoundException();
```

Comment:
iteration is inefficient here. Use `userDatabase.get()` directly and handle its exceptions